doc/develop/gui.rst

```diff
@@ -0,0 +1,145 @@
+************************************
+Graphical User Interface (GUI) Layer
+************************************
+
+Introduction
+============
+
+The Odemis graphical user interface (GUI) is implemented based on the Hierarchical Model–view–controller architectural pattern. (`Wikipedia <https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller>`_.) It consists of several main controllers and subcontrollers, each related to a specific aspect of Odemis functionality. This data is stored as vigilant attributes, so that the state can be subscribed to by components to notify them when changes occur.
+
+wxPython is used as an underlying framework for the GUI. Various resources are defined via XRC. Many of the stock wxWidget components have been extended in Odemis to better suit Odemis-specific use cases. 
+
+Models
+======
+
+When a session of the GUI loads, configuration and components from the backend are preloaded into a data structure representing the model, or shared state, of the GUI. This object, inherited from *MainGUIData*, is defined in ``/gui/model/__init__.py``, and contains attributes to all components loaded in the current microscope configuration. This object is accessible from many GUI components as the argument *main_data*.
+
+*MicroscopyGUIData* corresponds to a specific GUI tab, and contains data related to the shared state of that tab. 
+
+	- **LiveViewGUIData(MicroscopyGUIData):**
+	
+		Represents an interface used to only show the current data from the microscope.
+	
+	- **SparcAcquisitionGUIData(MicroscopyGUIData):**
+	
+		Represents an interface used to select a precise area to scan and
+		acquire signal. It allows fine control of the shape and density of the scan. 		It is specifically made for the SPARC system.
+
+	- **ChamberGUIData(MicroscopyGUIData):**
+	
+		Represents an interface used by the chamber view tab. 
+		
+	- **AnalysisGUIData(MicroscopyGUIData):**
+	
+		Represents an interface used to show the recorded microscope data. Typically
+		it represents all the data present in a specific file. All the streams should be StaticStreams
+		
+	- **ActuatorGUIData(MicroscopyGUIData):**
+	
+		Represents an interface used to move the actuators of a microscope. It might
+		also display one or more views, but it's not required. Typically used for the SECOM and SPARC(v2) alignment tabs.
+		
+	- **SecomAlignGUIData**, **SparcAlignGUIData**, and **Sparc2AlignGUIData**:
+	
+		Represents an interface used for alignment tabs. 
+
+
+Controllers
+===========
+
+.. figure:: gui_tree.*
+    :width: 100 %
+    :align: center
+
+The Odemis GUI has several main controllers that allow a user to directly control functionality. 
+
+The top level controllers include:
+    1. **TabBarCont** (``/gui/cont/tabs.py``)
```

Comment:
Please spell out completely the class names. `TabBarController`, `MenuController`.
Here, you could also mention the `SnapshotController` (in gui.cont.acquisition).